src/context/user.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useState } from "react";
+import { IProfileUserProps } from "../types/profileUser";
+
+export const UserProfileContext = createContext({} as any);
+
+
+export const UserProfileProvider = ({ children }: any) => {
+
+    const [user, setUser] = useState<IProfileUserProps>();
+    const [isStorage, setIsStorage] = useState(false);
+
+    const getStorageProfileUser = async () => {
+
+        const profileExist = await getProfileUserStored()
+
+        if (profileExist.login !== 'null') setIsStorage(true);
+    }
+
+    const setProfileUserLocalStored = (props: IProfileUserProps) => {        
+        localStorage.setItem("profileUser.id", props.id)
```

Comment:
in this method, instead of set each property separately, try to create an object that persist all this information. So when you set the object into the localStorage use the sintaxe:

```tsx
   const userInformation = {
     /* informations here*/
   };

  localStorage.setItem(JSON.stringify(userInformation));
```

You need to know that since you persist the object parsed by using `JSON.stringify()` method, when you use the object you need to parse again by using now the method `JSON.parse()`

[JSON parses](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)

Reply:
I tried this. But does'nt give right. Look under.This method await key and value
![image](https://user-images.githubusercontent.com/58925056/220635709-18912bc2-81a1-497c-a332-820ca56eab74.png)

Reply:
Sorry man, I commited a mistake. The right sintaxe is to declare a key to identify the values you are settings into the localStorage. Example:

```tsx
const userInformation = {
 /* informations here*/

localStorage.setItem("@your-key-to-identify-the-value", JSON.stringify(userInformation))
}
```